airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/response_to_file_extractor.py

```diff
@@ -0,0 +1,162 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+import logging
+import os
+import uuid
+import zlib
+from contextlib import closing
+from typing import Any, Dict, Iterable, Mapping, Optional, Tuple
+
+import pandas as pd
+import requests
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from numpy import nan
+
+EMPTY_STR: str = ""
+DEFAULT_ENCODING: str = "utf-8"
+DOWNLOAD_CHUNK_SIZE: int = 1024 * 1024 * 10
+
+
+class ResponseToFileExtractor(RecordExtractor):
+    """
+    This class is used when having very big HTTP responses (usually streamed) which would require too much memory so we use disk space as
+    a tradeoff.
+
+    Eventually, we want to support multiple file type by re-using the file based CDK parsers if possible. However, the lift is too high for
+    a first iteration so we will only support CSV parsing using pandas as salesforce and sendgrid were doing.
```

Comment:
do you have a plan for reconciling the file-based parsers with low-code?

Reply:
Not yet at all sources that could be declarative that I've checked (amazon-seller-partner, salesforce, sendgrid) are reading CSV files. Hence, this helps us push back the decision. In any case, I'm confident in the interfaces in the async jobs Python code and declarative interface (excluding the interpolation) so I'm not too worried

Reply:
imo there is value in reconciling the two so we don't need to maintain two csv parsers. This should probably be part of the Q3B follow up